File: c++/instruccion.cc
#include "Instruccion.h"
#include <string>

using namespace std;


Instruccion::Instruccion(){
}
//Cada clase add, write, etc. tendrá su propia implementación de estos métodos (por eso el virtual)
void ejecutar(stack<int> &pila);
int modificar(int pc);
string mostrar();

File: java/Instruccion.java
import java.util.Stack;

abstract class Instruccion {

    public Instruccion(){ // Constructor vacío 
    };
    
    public abstract void ejecutar(Stack<Integer> pila); // Lo defino como abstracto para indicar que las clases que hereden de esta clase deben implementar este método
    public abstract int incrementarPc(int pc); // Lo defino como abstracto para indicar que las clases que hereden de esta clase deben implementar este método
    public abstract String mostrar(); // Para poder listar el programa, cada instrucción debe poder devolver una cadena de texto que representa la propia instruccion 
}
